dotnet/samples/KernelSyntaxExamples/Example99_ApiManifest.cs

```diff
@@ -0,0 +1,190 @@
+    [Fact]
+    public async Task RunAsync()
+    {
+        string apiKey = TestConfiguration.AzureOpenAI.ApiKey;
+        string chatDeploymentName = TestConfiguration.AzureOpenAI.ChatDeploymentName;
+        string chatModelId = TestConfiguration.AzureOpenAI.ChatModelId;
+        string endpoint = TestConfiguration.AzureOpenAI.Endpoint;
+
+        if (apiKey == null || chatDeploymentName == null || chatModelId == null || endpoint == null)
+        {
+            this.WriteLine("Azure endpoint, apiKey, deploymentName, or modelId not found. Skipping example.");
+            return;
+        }
+
+        s_kernel = Kernel.CreateBuilder()
+            .AddAzureOpenAIChatCompletion(
+                deploymentName: chatDeploymentName,
+                endpoint: endpoint,
+                serviceId: "AzureOpenAIChat",
+                apiKey: apiKey,
+                modelId: chatModelId)
+            .Build();
+
+        s_sampleIndex = 1;
+
+        await RunSampleWithPlannerAsync(
+            "show the subject of my first message",
+            "latest email message subject is shown",
+            "MessagesPlugin");
+        await RunSampleWithPlannerAsync("get contents of file with id=test.txt",
+            "test.txt file is fetched and the content is shown",
+            "DriveItemPlugin",
+            "MessagesPlugin");
+        await RunSampleWithPlannerAsync(
+            "get contents of file with id=test.txt",
+            "test.txt file is not fetched because DriveItemPlugin is not loaded",
+            "MessagesPlugin");
+        await RunSampleWithPlannerAsync(
+            "tell me how many contacts I have",
+            "number of contacts is shown",
+            "MessagesPlugin",
+            "ContactsPlugin");
+        await RunSampleWithPlannerAsync(
+            "tell me title of first event in my calendar",
+            "title of first event from calendar is shown if exists",
+            "MessagesPlugin",
+            "CalendarPlugin");
+    }
+
+    private void WriteSampleHeadingToConsole(string goal, string expectedOutputDescription, params string[] pluginNames)
+    {
+        this.WriteLine();
+        this.WriteLine($"======== [ApiManifest Plugins] Sample {s_sampleIndex++} - Create and Execute \"{goal}\" Plan ========");
+        this.WriteLine($"======== Plugins: {string.Join(" ", pluginNames)} ========");
+        this.WriteLine($"======== Expected Output: {expectedOutputDescription} ========");
+        this.WriteLine();
+    }
+
+    private async Task RunSampleWithPlannerAsync(string goal, string expectedOutputDescription, params string[] pluginNames)
+    {
+        _ = s_kernel ?? throw new KernelException("Kernel not initialized!");
+        WriteSampleHeadingToConsole(goal, expectedOutputDescription, pluginNames);
+        s_kernel.Plugins.Clear();
+        await AddApiManifestPluginsAsync(pluginNames).ConfigureAwait(false);
+
+        var result = await s_planner.ExecuteAsync(s_kernel, goal);
```

Comment:
Thank you @SergeyMenshykh!

The reason we wanted to have these samples was to be able to create feedback to API developers and APIManifest spec developers so that they can tailor for the world of LLMs, i.e. how can they reshape APIs and how much additional guidance is needed on top of regular OpenAPI spec to guide LLMs better and more deterministically. So the nature of these tests is more e2e than unit testing. That said, I understand that KernelSyntaxExamples may not be the right place. We plan to continuously watch for how well the manifest works with the planners. If KernelSyntaxExamples is not the right place for that, I'd be happy to hear your guidance on the alternatives.

Reply:
for now, we will maintain a separate console application for ApiManifest demos with planners, converted KernelSyntaxExamples tests to directly call the plugin functions.